### ISSUE TITLE
Split `README` in multiple files

### ISSUE THREAD

**Issue by mr-archano** [novoda/gradle-build-properties-plugin#21]
The README as it is now is very long and quite confusing. A new user should just be able to glance at it and understand how to use the plugin. Anything else can be covered as separate documentation (separate markdown documents in the repo).

**Comment by ataulm** [comment]
I think this is enough to get started. It shows what is (for me) the two common scenarios: using the values in your build script, and using the values to generate fields in BuildConfig. I tried to use variations in the naming to make it clear that the keys did not have to match a convention, nor that the file names had to match the `BuildProperties` handle.

I have loads of questions though - maybe we can start from here, and put the rest into `/docs/README.md` with a link, then separate out sections (probably one page is still enough) from there?

---

# gradle-build-properties-plugin
[![](https://ci.novoda.com/buildStatus/icon?job=gradle-build-properties-plugin)](https://ci.novoda.com/job/gradle-build-properties-plugin/lastSuccessfulBuild/console) [![](https://img.shields.io/badge/License-Apache%202.0-lightgrey.svg)](LICENSE.txt) [![Bintray](https://api.bintray.com/packages/novoda/maven/gradle-build-properties-plugin/images/download.svg) ](https://bintray.com/novoda/maven/gradle-build-properties-plugin/_latestVersion)

External properties files support for your build scripts.

## Description

Manage properties you want to keep out of version control (e.g. keys, credentials, endpoints etc.). This plugin simplifies:

- accessing properties files from your build script
- generating fields in Android's `BuildConfig` class with values from your properties files
- generating Android resources with values from your properties files

## Adding to your project

The plugin is available from JCenter:

```groovy
buildscript {
  repositories {
    jcenter()
  }
  dependencies {
    classpath 'com.novoda:gradle-build-properties-plugin:<latest-version>'
  }
}
```

Apply the plugin in your build script:

```groovy
apply plugin: 'com.novoda.build-properties'
```

## Simple usage

Given the following properties file, `myproject/properties/version.properties`:

```
VERSION_CODE=123
VERSION_NAME="1.2.3"
```

you can create a reference to it in your build script, `myproject/app/build.gradle`:

```groovy
buildProperties {
    versions {
        file project.file('../properties/version.properties')
    }
}

android {
    ...

    defaultConfig {
        versionCode buildProperties.versions['VERSION_CODE'].int
        versionName buildProperties.versions['VERSION_NAME'].string
        ...
```

You can create multiple properties files. `myproject/properties/api_secrets.properties`:

```
api_client_id="abcdefg"
api_client_secret="1234567"
```

and you'll just need to add a handle to it (can be different from the file name) in the `buildProperties` closure:

```groovy
buildProperties {
    versions {
        file project.file('../properties/version.properties')
    }
    api {
        file project.file('../properties/api_secrets.properties')
    }
}

android {
    ...

    defaultConfig {
        buildConfigString 'CLIENT_ID', buildProperties.api['api_client_id']
        buildConfigString 'CLIENT_SECRET', buildProperties.api['api_client_secret']
        ...
```

**Comment by ataulm** [comment]
Here are some of the questions:

- in the versions example, if I didn't postfix with `.int` it doesn't work (didn't, about a month ago). why?
- in the versions example, if I didn't postfix, would it default to string? even though it's not wrapped in `""` in the properties file?
- is the `buildProperties` handle something defined by the plugin? OR is it a reference to the `buildProperties` closure which is defined (required) by the plugin?
- for the resource creation, is it just string resources? Or integer, dimen, color etc too?

**Comment by mr-archano** [comment]
> in the versions example, if I didn't postfix with .int it doesn't work (didn't, about a month ago). why?
* `android.defaultConfig.versionCode` expects `int` as input so you have to use `.int` to force the evaluation as such type (see [`BuildProperties#getAt(String)`](https://github.com/novoda/gradle-build-properties-plugin/blob/master/plugin/src/main/groovy/com/novoda/buildproperties/BuildProperties.groovy#L42-L44))
> in the versions example, if I didn't postfix, would it default to string? even though it's not wrapped in "" in the properties file?
* It depends uniquely where are you using the entry: in the example above `versionCode` and `versionName` can't understand what an `Entry` is, therefore you have to evaluate that (via `.int` or `.string`). If instead you use one of the utilities provided by the plugin that extends the basic Android plugin DSL (ie: `buildConfigInt`, `buildConfigString`, etc) then you can save the type casting because they support also `Entry` as input (since #23)
>  is the buildProperties handle something defined by the plugin? OR is it a reference to the buildProperties closure which is defined (required) by the plugin?
* It's a [`NamedDomainObjectContainer<BuildProperties>`](https://docs.gradle.org/3.5/javadoc/org/gradle/api/NamedDomainObjectContainer.html), basically a container defined [by the plugin](https://github.com/novoda/gradle-build-properties-plugin/blob/master/plugin/src/main/groovy/com/novoda/buildproperties/BuildPropertiesPlugin.groovy#L11-L17) holding `BuildProperties` (à la `androidConfig.signingConfig`, `android.productFlavors`, etc).
>  for the resource creation, is it just string resources? Or integer, dimen, color etc too?
* As mentioned in the [current README](https://github.com/novoda/gradle-build-properties-plugin/blob/master/README.md#3-typed-buildconfigfield--resvalue) the plugin provides only support for int, boolean and string on top of `resValue`. Other types can be added I suppose.

**Comment by mr-archano** [comment]
Addressed by #49
